validator/client/src/test/java/tech/pegasys/teku/validator/client/ValidatorRegistratorTest.java

```diff
@@ -100,63 +111,50 @@ void registersValidators_onBeginningOfEpoch() {
     when(ownedValidators.getActiveValidators())
         .thenReturn(List.of(validator1, validator2, validator3));
 
-    doAnswer(invocation -> SafeFuture.completedFuture(dataStructureUtil.randomSignature()))
-        .when(signer)
-        .signValidatorRegistration(any(ValidatorRegistration.class), any(UInt64.class));
+    when(feeRecipientProvider.getFeeRecipient(any())).thenReturn(Optional.of(eth1Address));
 
     // WHEN
     validatorRegistrator.onSlot(UInt64.ZERO);
 
-    // THEN
-    @SuppressWarnings("unchecked")
-    ArgumentCaptor<SszList<SignedValidatorRegistration>> argumentCaptor =
-        ArgumentCaptor.forClass(SszList.class);
-
-    InOrder inOrder = Mockito.inOrder(validatorApiChannel);
-
-    inOrder.verify(validatorApiChannel, timeout(1000)).registerValidators(argumentCaptor.capture());
-    verifyRegistrations(argumentCaptor.getValue());
+    // Make sure all the futures have completed before next call
+    verify(validatorApiChannel, timeout(1000)).registerValidators(any());
```

Comment:
Not sure if you really need these timeout() since you provide completed futures, so everything should be synchronous

Reply:
Agree, removed the timeouts